### DISCUSSION TITLE
Script evaluated within newly added child components doesn't have access to the component

### DISCUSSION THREAD

**Discussion by cristiancalara** [livewire/livewire#3019] [7. Bugs (archived)]
### Description
@this directive in a script tag in a dynamically added child component throws an error. Not sure if this is a bug? Couldn't find anything in the docs about it.

### Stripped-down, copy-pastable code snippets
```php
class ParentComponent extends Component
{
    public $showChild = false;

    public function showChild()
    {
        $this->showChild = true;
    }

    public function render()
    {
        return view('livewire.parent-component');
    }
}
```
```blade
<div>
	<button wire:click="showChild">
		Show child
	</button>
	@if($showChild)
		<livewire:child-component/>
	@endif
</div>
```
```php
class ChildComponent extends Component
{
    public function render()
    {
        return view('livewire.child-component');
    }
}
```
```blade
<div>
	<span>Child component</span>
         // This will throw a "TypeError: Cannot read property '$wire' of undefined"
	<script>console.log(@this)</script> 
</div>
```

### Exact steps to reproduce
After clicking the "Show child" button in the parent component, the `console.log(@this)` will throw a "TypeError: Cannot read property '$wire' of undefined". After debugging a bit, it seems that the script is actually evaluated before the component is added to the store i.e. `nodeInitializer.initialize(el, this)` is called before `store.addComponent(new Component(el, this.connection))`  https://share.getcloudapp.com/P8u5zrnL  

**Context**
- Livewire version: 2.4.1
- Laravel version: 8.35.1
- Alpine version: 2.7.3
- Browser: All

Thanks for the nice work on Livewire!

**Comment by joshhanley** [comment]
@cristiancalara yeah that issue was fixed in Alpine 2.8.2.

Can you please upgrade it and let us know how it goes?

Edit: Sorry just realise you weren't using Alpine in the example. I'd still recommend upgrading Livewire to latest version 2.4.4 and Alpine to 2.8.2 and test again :)

**Comment by cristiancalara** [comment]
Hey @joshhanley - thanks for the quick reply and sorry for not upgrading to the latest versions before opening the issue.

I've now updated to the latest versions (both Livewire and Alpine). The issue still happens.
